### PR TITLE
Generate cluster monitoring config

### PR DESCRIPTION
**Pull request by NautiluX**
* fix and avoid drift between UWM and non-UWM clusters

Jira: [OSD-12572](https://issues.redhat.com//browse/OSD-12572)

### What type of PR is this?
_(bug/feature/cleanup/documentation)_

### Which Jira/Github issue(s) this PR fixes?

_Fixes #_

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Included documentation changes with PR
- [ ] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

### CODE REVIEW

**Comment by bdematte** [comment]
It could be great to make a note on the global README as well (to make awareness of the possiblity of generating resource through script and the option if we have other usecases to make the script customizable in the future rather than hard-coding values in the script) .

File: hack/00-osd-managed-cluster-config-integration.yaml.tmpl [hunk 1]
```diff
@@ -24651,40 +24651,36 @@ objects:
...
-          \ NoSchedule\n      key: node-role.kubernetes.io/infra\n      operator:\
-          \ Exists\n  retention: 11d\n  retentionSize: 90GB\n  volumeClaimTemplate:\n\
+        config.yaml: "enableUserWorkload: true\nprometheusK8s:\n  remoteWrite:\n \
+          \ - url: http://token-refresher.openshift-monitoring.svc.cluster.local\n\
+          \    remoteTimeout: 30s\n    writeRelabelConfigs:\n    - sourceLabels:\n\
+          \      - __name__\n      action: keep\n      regex: (addon_operator_addons_count|addon_operator_addon_health_info|addon_operator_ocm_api_requests_durations|addon_operator_ocm_api_requests_durations_sum|addon_operator_ocm_api_requests_durations_count|addon_operator_paused|cluster_admin_enabled|limited_support_enabled|identity_provider|ingress_canary_route_reachable|ocm_agent_service_log_sent_total|sre:slo:probe_success_api|sre:slo:probe_success_console|sre:slo:upgradeoperator_upgrade_result|sre:slo:imageregistry_http_requests_total|sre:slo:oauth_server_requests_total|sre:sla:outage_5_minutes|sre:slo:apiserver_28d_slo|sre:slo:console_28d_slo|sre:error_budget_burn:apiserver_28d_slo|sre:error_budget_burn:console_28d_slo)\n\
```
Comment (bdematte) [inline hack/00-osd-managed-cluster-config-integration.yaml.tmpl:24657]: Isn't the missing single-quote troublesome for the proper behaviour ?

Comment (NautiluX) [reply hack/00-osd-managed-cluster-config-integration.yaml.tmpl:24657]: I think it gets lost in reading the yaml with pyyaml and then dumping it to another place so I _think_ it should be fine.
`yq` is happy with it, at least with the input file that doesn't have the quote either:
```
$ yq '.data."config.yaml"' deploy/cluster-monitoring-config-non-uwm/50-GENERATED-cluster-monitoring-config.yaml | yq .prometheusK8s.remoteWrite[0].writeRelabelConfigs[0].regex
(addon_operator_addons_count|addon_operator_addon_health_info|addon_operator_ocm_api_requests_durations|addon_operator_ocm_api_requests_durations_sum|addon_operator_ocm_api_requests_durations_count|addon_operator_paused|cluster_admin_enabled|limited_support_enabled|identity_provider|ingress_canary_route_reachable|ocm_agent_service_log_sent_total|sre:slo:probe_success_api|sre:slo:probe_success_console|sre:slo:upgradeoperator_upgrade_result|sre:slo:imageregistry_http_requests_total|sre:slo:oauth_server_requests_total|sre:sla:outage_5_minutes|sre:slo:apiserver_28d_slo|sre:slo:console_28d_slo|sre:error_budget_burn:apiserver_28d_slo|sre:error_budget_burn:console_28d_slo)
```

**Pull request merged**